nebula_ros/src/continental/continental_ars548_decoder_ros_wrapper.cpp

```diff
@@ -0,0 +1,344 @@
+
+  RCLCPP_INFO_STREAM(this->get_logger(), "SensorConfig:" << sensor_configuration);
+  return Status::OK;
+}
+
+void ContinentalARS548DriverRosWrapper::DetectionListCallback(
+  std::unique_ptr<continental_msgs::msg::ContinentalArs548DetectionList> msg)
+{
+  if (
+    detection_pointcloud_pub_->get_subscription_count() > 0 ||
+    detection_pointcloud_pub_->get_intra_process_subscription_count() > 0) {
+    const auto detection_pointcloud_ptr = ConvertToPointcloud(*msg);
+    auto detection_pointcloud_msg_ptr = std::make_unique<sensor_msgs::msg::PointCloud2>();
+    pcl::toROSMsg(*detection_pointcloud_ptr, *detection_pointcloud_msg_ptr);
+
+    detection_pointcloud_msg_ptr->header = msg->header;
+    detection_pointcloud_pub_->publish(std::move(detection_pointcloud_msg_ptr));
+  }
+  if (
+    detection_list_pub_->get_subscription_count() > 0 ||
+    detection_list_pub_->get_intra_process_subscription_count() > 0) {
+    detection_list_pub_->publish(std::move(msg));
+  }
+}
+
+void ContinentalARS548DriverRosWrapper::ObjectListCallback(
+  std::unique_ptr<continental_msgs::msg::ContinentalArs548ObjectList> msg)
+{
+  if (
+    object_pointcloud_pub_->get_subscription_count() > 0 ||
+    object_pointcloud_pub_->get_intra_process_subscription_count() > 0) {
+    const auto object_pointcloud_ptr = ConvertToPointcloud(*msg);
+    auto object_pointcloud_msg_ptr = std::make_unique<sensor_msgs::msg::PointCloud2>();
+    pcl::toROSMsg(*object_pointcloud_ptr, *object_pointcloud_msg_ptr);
+
+    object_pointcloud_msg_ptr->header = msg->header;
+    object_pointcloud_pub_->publish(std::move(object_pointcloud_msg_ptr));
+  }
+  if (
+    object_list_pub_->get_subscription_count() > 0 ||
+    object_list_pub_->get_intra_process_subscription_count() > 0) {
+    object_list_pub_->publish(std::move(msg));
+  }
+}
+
+pcl::PointCloud<pcl::PointXYZ>::Ptr ContinentalARS548DriverRosWrapper::ConvertToPointcloud(
+  const continental_msgs::msg::ContinentalArs548DetectionList & msg)
+{
+  pcl::PointCloud<pcl::PointXYZ>::Ptr output_pointcloud(new pcl::PointCloud<pcl::PointXYZ>);
+  output_pointcloud->reserve(msg.detections.size());
+
+  pcl::PointXYZ point{};
+  for (const auto & detection : msg.detections) {
+    point.x =
+      std::cos(detection.elevation_angle) * std::cos(detection.azimuth_angle) * detection.range;
+    point.y =
+      std::cos(detection.elevation_angle) * std::sin(detection.azimuth_angle) * detection.range;
+    point.z = std::sin(detection.elevation_angle) * detection.range;
+
+    output_pointcloud->points.emplace_back(point);
+  }
+
+  output_pointcloud->height = 1;
+  output_pointcloud->width = output_pointcloud->points.size();
+  return output_pointcloud;
+}
+
+pcl::PointCloud<pcl::PointXYZ>::Ptr ContinentalARS548DriverRosWrapper::ConvertToPointcloud(
+  const continental_msgs::msg::ContinentalArs548ObjectList & msg)
+{
+  pcl::PointCloud<pcl::PointXYZ>::Ptr output_pointcloud(new pcl::PointCloud<pcl::PointXYZ>);
```

Comment:
`(optional)(nice-to-have)` I am not familiar with radar visualization workflows but from my understanding, the pointclouds are only output as a visualization aid.
When visualizing the clouds in Rviz, it was quite hard to figure out what is noise and what are actual detections.
Thus, provided that people will actually use Rviz to visualize radar output, it would be nice to have more information per point (especially given that `...Detection.msg` and `...Object.msg` contain so much visualizable data).

Reply:
@mojomex 
In autoware we have pipelines for this but they are probably overfited for the ARS408 so I actually need to talk with the people involved later (mostly in perception from what I understand).

For example, ways to do this are:
 - We create our own markers for rviz in nebula
 - We create / modify the rviz plugins in autoware to accept directly the new radar messages and be able to have more control over the visualization than just markers